onedal/basic_statistics/basic_statistics.py

```diff
@@ -85,27 +84,18 @@ def _compute(self, data, weights, module, queue):
             weights = np.asarray(weights)
 
         data, weights = _convert_to_supported(policy, data, weights)
-
         data_table, weights_table = to_table(data, weights)
 
         dtype = data.dtype
-        res = self._compute_raw(data_table, weights_table, module, policy, dtype)
-
-        return {k: from_table(v).ravel() for k, v in res.items()}
+        raw_result = self.compute_raw(data_table, weights_table, policy, dtype)
+        for opt, value in raw_result.items():
+            setattr(self, opt, from_table(value).ravel())
 
-
-class BasicStatistics(BaseBasicStatistics):
-    """
-    Basic Statistics oneDAL implementation.
-    """
-
-    def __init__(self, result_options="all", *, algorithm="by_default", **kwargs):
-        super().__init__(result_options, algorithm)
-
-    def compute(self, data, weights=None, queue=None):
-        return super()._compute(data, weights, _backend.basic_statistics.compute, queue)
+        return self
 
     def compute_raw(self, data_table, weights_table, policy, dtype=np.float32):
-        return super()._compute_raw(
-            data_table, weights_table, _backend.basic_statistics.compute, policy, dtype
-        )
+        params = self._get_onedal_params(dtype)
+        result = self._module.compute(policy, params, data_table, weights_table)
+        options = self._get_result_options(self.options).split("|")
+
+        return {opt: getattr(result, opt) for opt in options}
```

Comment:
Generally, for public methods it is better to have defaults values for all params, except input data.

But for this function it will be hard to provide public iface, because for user it would very complicated to define policy value and tables.

It is better to leave it as is private, since it is only internally used.